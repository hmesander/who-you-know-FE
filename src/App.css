.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #283e4a;
  height: 30px;
  padding: 20px;
  display: flex;
  color: #fff;
  justify-content: flex-start;
}

.App-title {
  font-size: 1em;
}

.App-intro {
  font-size: large;
}

.login-button {
  color: #fff;
  height: 34px;
  line-height: 32px;
  width: 200px;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: bold;
  background-color: #f6e312;
  border: 1px solid #e9ac1a;
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
  border-radius: 3px;
  color: #333;
  cursor: pointer;
  box-sizing: border-box;
  background-image: linear-gradient(to bottom, #f6e312 0%,#f9c80d 100%);
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
}

.login-button:hover {
  background-image: linear-gradient(to bottom, #eddb11 0%,#efcf14 35%,#f1bf18 65%,#f2b21b 100%);
  box-shadow: 0 1px 3px rgba(0,0,0,0.25);
}