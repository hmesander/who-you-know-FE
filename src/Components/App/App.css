.App {
  text-align: center;
  /* font-family: Sans-Serif; */
}

body {
  background-color: #e7e7e7;
}

.App-header {
  background-color: #283e4a;
  height: 30px;
  padding: 20px;
  display: flex;
  color: #fff;
  justify-content: space-between;
  align-items: center;
}

.App-title {
  font-size: 1.5em;
  font-weight: 600;
  background-color: none;
}

.logo {
  background-color: #0077b5;
  border-radius: 2px;
  height: 50px;
  margin: 0 4px;
  letter-spacing: -1px;
  padding: 3px;
  text-align: center;
}

.App-intro {
  font-size: 1em;
  font-weight: 100;
  margin-right: 25px;
}

.login-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.buttons {
  background-color: #0073b1;
  border: 0;
  border-radius: 2px;
  color: #fff;
  cursor: pointer;
}

.buttons:hover {
  background-color: #006097;
}

.login-button, .logout-button {
  height: 24px;
  width: 100px;
}

.sort-cards-button, .play-deck-button {
  line-height: 1.2rem;
  width: 13em;
  font-size: 0.8rem;
  font-weight: 400;
  padding: 4px 15px;
} 

.controls-component {
  width: 60%;
  margin: 5% auto;
  display: flex;
  justify-content: space-evenly;
}

.login-error-container {
  width: 50%;
  margin: 2em auto;
  text-align: center;
}

.login-error-text {
  font-size: 1em;
}