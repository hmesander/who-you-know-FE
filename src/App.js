import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import Controls from './Controls.js';
import Login from './Login.js';
import Sort from './Sort.js';
import './App.css';
import { authorizeUser } from './helpers/apiCalls.js';

class App extends Component {
  constructor() {
    super();

  }

  handleLogin = () => {
    console.log('click');
    this.props.history.push('/welcome');
  }

  handleSort = () => {
    console.log('sort');
    this.props.history.push('/sort');
  }

  handlePlayDeck = () => {
    console.log('play');
    this.props.history.push('/play');
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Who 
          you
          know.</h1>
        </header>
        <p className="App-intro">
          To get started, please login.
        </p>
        <button 
          className="login-button" 
          onClick={this.handleLogin}>
          Login
        </button>
        {/* <Route exact path='/' component={Login} /> */}
        <Route exact path='/welcome/' render={() => <Controls handleSort={this.handleSort} handlePlay={this.handlePlayDeck} />} />
        <Route path='/sort/' render={() => <Sort />} />
      </div>
    );
  }
}

export default withRouter(App);
