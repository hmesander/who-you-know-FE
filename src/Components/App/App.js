import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import Controls from '../Controls/Controls.js';
import Sort from '../Sort/Sort.js';
import PlayCards from '../PlayCards/PlayCards.js';
import './App.css';
import Auth from '../../Auth/Auth.js';
import history from '../../history';
import PropTypes from 'prop-types';
import { postTokens } from '../../helpers/apiCalls.js';
const auth = new Auth();

export class App extends Component {
  constructor() {
    super();

    this.state = {
      hash: '',
      idToken: '',
      accessToken: '',
      userId: ''
    };
  }

  async componentDidMount() {
    if (history.location.hash) {
      await this.setState({
        hash: history.location.hash
      });
      let { idToken, accessToken } = await this.parseURL(history.location.hash);
      this.setState({
        idToken,
        accessToken
      });
      let results = await postTokens(this.state.accessToken, this.state.idToken);
      this.setState({
        userId: results.id
      });
    } 
  }

  handleLogin = () => {
    auth.login();
  }

  handleLogout = () => {
    auth.logout();
    this.setState({
      hash: '', 
      userId: ''
    });
    this.props.history.push('/');
  }

  handleSort = () => {
    this.props.history.push('/sort');
  }

  handlePlayDeck = () => {
    this.props.history.push('/play');
  }

  parseURL = url => {
    let splitUrl = url.split('#access_token=');
    if (splitUrl.length > 1) {
      let accessToken = splitUrl[1].split('&expires_in=')[0];
      let idToken = splitUrl[1].split('&id_token=')[1];
      return { idToken, accessToken };
    }
  }

  render() {
    let display;
    if (this.state.hash !== '' || this.state.userId) {
      display = 
      <button
        className="buttons logout-button"
        onClick={this.handleLogout}>
          Logout
      </button>;
    } else {
      display = 
      <div className="login-container">
        <p className="App-intro">
          To get started, please login.
        </p>
        <button
          className="buttons login-button"
          onClick={this.handleLogin}>
          Login
        </button>
      </div>;
    }

    return (
      <div className="App">
        <header className="App-header">
          <div className='title-container'>
            <h1 className="App-title">Who 
              <span className='logo'>you</span> 
          know.</h1>
          </div>
          
          {display}
        </header>
        <Route path='/callback' render={() => <Controls handleSort={this.handleSort} handlePlay={this.handlePlayDeck} />} />
        <Route exact path='/sort' render={() => <Sort userId={this.state.userId} handlePlay={this.handlePlayDeck}/>} />
        <Route exact path='/play' render={() => <PlayCards userId={this.state.userId}/>} />
      </div>
    );
    
  }
}
export default withRouter(App);

App.propTypes = {
  history: PropTypes.string
};
