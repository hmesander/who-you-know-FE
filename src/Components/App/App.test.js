/*eslint-disable no-undef*/

import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';


describe('APP TESTS', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<App history={''} />, { disableLifecycleMethods: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should display a login button if the user is not logged in', () => {
    const wrapper = shallow(<App history={''}/>, { disableLifecycleMethods: true });
    
    expect(wrapper.find('.login-button').length).toEqual(1);
  });

  it('should display a logout button if the user is logged in', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    wrapper.instance().setState({hash: 'taco'});
    
    expect(wrapper.find('.logout-button').length).toEqual(1);
  });

  it('calls the login function when a user clicks the login button', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    const spy = spyOn(wrapper.instance(), 'handleLogin');
    wrapper.instance().forceUpdate();

    wrapper.find('.login-button').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it('should call the logout function when a user clicks the logout button', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    wrapper.instance().setState({ hash: 'taco' });
    const spy = spyOn(wrapper.instance(), 'handleLogout');
    wrapper.instance().forceUpdate();

    wrapper.find('.logout-button').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it('should reset hash in state to an empty string when handleLogout is called', () => {
    const wrapper = shallow(<App history={[]}/>, { disableLifecycleMethods: true });
    wrapper.instance().setState({
      hash: 'taco'
    });
    wrapper.instance().handleLogout();

    expect(wrapper.instance().state.hash).toEqual('');
  });

  it('should push the history object to sort when handleLogout is called', () => {
    const wrapper = shallow(<App history={[]} />, { disableLifecycleMethods: true });
    wrapper.instance().setState({
      hash: 'taco'
    });
    wrapper.instance().handleLogout();

    expect(wrapper.instance().props.history).toEqual(['/']);
  });

  it('should parse a url and returns an object with the tokens when parseUrl is called', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    const mockUnparsedURL = '#access_token=taco&expires_in=7200&token_type=Bearer&state=cheeseburgers&id_token=123456789';
    wrapper.setState({hash: mockUnparsedURL});
    const results = wrapper.instance().parseURL(mockUnparsedURL);

    expect(results.accessToken).toEqual('taco');
    expect(results.idToken).toEqual('123456789');
  });

  it('should push the history object to sort when handleSort is called', () => {
    const wrapper = shallow(<App history={[]} />, { disableLifecycleMethods: true });
    wrapper.instance().handleSort();
    expect(wrapper.instance().props.history).toEqual(['/sort']);
  });

  it('should push the history object to play when handlePlayDeck is called', () => {
    const wrapper = shallow(<App history={[]} />, { disableLifecycleMethods: true });
    wrapper.instance().handlePlayDeck();
    expect(wrapper.instance().props.history).toEqual(['/play']);
  });
});