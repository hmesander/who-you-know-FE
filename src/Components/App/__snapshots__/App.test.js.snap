// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP TESTS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    history=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <div
                    className="title-container"
          >
                    <h1
                              className="App-title"
                    >
                              Who
                              <span
                                        className="logo"
                              >
                                        you
                              </span>
                              know.
                    </h1>
          </div>
          <div
                    className="login-container"
          >
                    <p
                              className="App-intro"
                    >
                              To get started, please login.
                    </p>
                    <button
                              className="buttons login-button"
                              onClick={[Function]}
                    >
                              Login
                    </button>
          </div>
</header>,
        <Route
          path="/callback"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/sort"
          render={[Function]}
/>,
        <Route
          exact={true}
          path="/play"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-container"
>
              <h1
                            className="App-title"
              >
                            Who
                            <span
                                          className="logo"
                            >
                                          you
                            </span>
                            know.
              </h1>
</div>,
            <div
              className="login-container"
>
              <p
                            className="App-intro"
              >
                            To get started, please login.
              </p>
              <button
                            className="buttons login-button"
                            onClick={[Function]}
              >
                            Login
              </button>
</div>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="App-title"
>
                Who
                <span
                                className="logo"
                >
                                you
                </span>
                know.
</h1>,
              "className": "title-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Who",
                  <span
                    className="logo"
>
                    you
</span>,
                  "know.",
                ],
                "className": "App-title",
              },
              "ref": null,
              "rendered": Array [
                "Who",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "you",
                    "className": "logo",
                  },
                  "ref": null,
                  "rendered": "you",
                  "type": "span",
                },
                "know.",
              ],
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="App-intro"
>
                  To get started, please login.
</p>,
                <button
                  className="buttons login-button"
                  onClick={[Function]}
>
                  Login
</button>,
              ],
              "className": "login-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "To get started, please login.",
                  "className": "App-intro",
                },
                "ref": null,
                "rendered": "To get started, please login.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "buttons login-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/callback",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/sort",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/play",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <div
                        className="title-container"
            >
                        <h1
                                    className="App-title"
                        >
                                    Who
                                    <span
                                                className="logo"
                                    >
                                                you
                                    </span>
                                    know.
                        </h1>
            </div>
            <div
                        className="login-container"
            >
                        <p
                                    className="App-intro"
                        >
                                    To get started, please login.
                        </p>
                        <button
                                    className="buttons login-button"
                                    onClick={[Function]}
                        >
                                    Login
                        </button>
            </div>
</header>,
          <Route
            path="/callback"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/sort"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/play"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-container"
>
                <h1
                                className="App-title"
                >
                                Who
                                <span
                                                className="logo"
                                >
                                                you
                                </span>
                                know.
                </h1>
</div>,
              <div
                className="login-container"
>
                <p
                                className="App-intro"
                >
                                To get started, please login.
                </p>
                <button
                                className="buttons login-button"
                                onClick={[Function]}
                >
                                Login
                </button>
</div>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="App-title"
>
                  Who
                  <span
                                    className="logo"
                  >
                                    you
                  </span>
                  know.
</h1>,
                "className": "title-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Who",
                    <span
                      className="logo"
>
                      you
</span>,
                    "know.",
                  ],
                  "className": "App-title",
                },
                "ref": null,
                "rendered": Array [
                  "Who",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "you",
                      "className": "logo",
                    },
                    "ref": null,
                    "rendered": "you",
                    "type": "span",
                  },
                  "know.",
                ],
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="App-intro"
>
                    To get started, please login.
</p>,
                  <button
                    className="buttons login-button"
                    onClick={[Function]}
>
                    Login
</button>,
                ],
                "className": "login-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "To get started, please login.",
                    "className": "App-intro",
                  },
                  "ref": null,
                  "rendered": "To get started, please login.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "buttons login-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/callback",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/sort",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/play",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
